standards/security_standards.md

```diff
@@ -0,0 +1,440 @@
+
+3.4.7. Deny by Default. Deny by default is the principle that if a request is not specifically allowed, it is denied.
+
+3.4.8. Don’t hardcode roles.
+
+3.4.9. Enforce authorization controls on every request, including those made by server-side scripts, "includes" and requests from rich client-side technologies.
+
+3.4.10. Restrict access to protected URLs to only authorized users.
+
+3.4.11. Restrict access to protected functions to only authorized users.
+
+3.5. Validation, Sanitization and Encoding
+
+The application must satisfy the following high-level requirements:
+
+- Input data is strongly typed, validated, range or length checked, or at worst, sanitized or filtered.
+- Output data is encoded or escaped as per the context of the data as close to the interpreter as possible.
+
+3.5.1. Input Validation
+
+3.5.1.1. Minimum and maximum value ranges must be checked for numerical parameters and dates, and minimum and maximum lengths must be checked for strings.
+
+3.5.1.2. Add validation against JSON Schema and XML Schema (XSD) for input in these formats.
+
+3.5.1.3. Ensure that any input validation performed on the client is also performed on the server.
+
+3.5.1.4. Allow list validation is the appropriate validation for all input fields provided by the user. Allow list
+
+validation involves defining exactly what IS
+
+authorized, and by definition, everything else is not
+
+authorized. If it's well-structured data, like dates,
+
+social security numbers, zip codes, email
+
+addresses, etc. then the developer should be able
+
+to define a very strong validation pattern, usually
+
+based on regular expressions, for validating such
+
+input. If the input field comes from a fixed set of
+
+options, like a drop-down list or radio buttons, then
+
+the input needs to match exactly one of the values
+
+offered to the user in the first place.
+
+3.5.1.5. All languages and most frameworks provide validation libraries or functions that must be leveraged to validate data. Validation libraries typically cover common data types, length requirements, integer ranges, “is null” checks and more.
+
+3.5.1.6. Regular expressions offer a way to check whether data matches a specific pattern. When designing regular expressions, be aware of RegEx Denial of Service (ReDoS) attacks. These attacks cause a program using a poorly designed Regular Expression to operate very slowly and utilize CPU resources for a very long time.
+
+3.5.1.7. Must validate all client-provided data before processing, including all parameters, URLs and HTTP header content (e.g. Cookie names, User-agent and values).
+
+3.5.2. Sanitization and Sandboxing
+
+3.5.2.1. All untrusted HTML input from WYSIWYG editors or similar must be properly sanitized with an HTML sanitizer library or framework feature.
+
+3.5.2.2. The application must sanitize user input before passing to mail systems to protect against SMTP or IMAP injection.
+
+3.5.2.3. OWASP recommends DOMPurify for HTML Sanitization.
```

Comment:
Let's link this

Reply:
@JHNLWHD like to what topic specifically?

Reply:
The OWASP recommendation of DOMPurify for HTML Sanitization

Reply:
Done adding DOMPurify link